KTB-04-string-parse

DenisGubernatorov opened the pull request:
В ветке указан номер задачи, согласно порядку на сайте. Т.е. не KTB-02, если исходить из названия ветки для предыдущего ПР, а KTB-04

1) Добавлен data class Word, хранящий слово, его перевод и количество правильных ответов 2) Добавлен список dictionary, хранящий объекты типа Word 3) Получение значения для количества правильных ответов реализовано в отдельном методе getCorrectAnswersCountValue. Учитывается не только что значения может не быть, но и то, что может быть пустая строка 4) Заполнение dictionary.
5) Вывод в консоль элементов из dictionary

DenisGubernatorov committed:
KTB-04-string-parse
1) Добавлен data class Word, хранящий слово, его перевод и количество правильных ответов
2) Добавлен список dictionary, хранящий объекты типа Word
3) Получение значения для количества правильных ответов реализовано в отдельном методе getCorrectAnswersCountValue. Учитывается не только что значения может не быть, но и то, что может быть пустая строка
4) Заполнение dictionary.
5) Вывод в консоль элементов из dictionary

kaznikit commented on src/main/kotlin/Main.kt:
```diff
@@ -3,9 +3,25 @@ package org.example
+        val values = it.split("|")
+        val correctAnswersCount = getCorrectAnswersCountValue(values)
+
+        dictionary.add(Word(values[0], values[1], correctAnswersCount.toInt()))
```
зачем так усложнять функцию получения правильных ответов, проще будет сделать таким образом: `values.getOrNull(2)?.toIntOrNull() ?: 0`

kaznikit commented:
> В ветке указан номер задачи, согласно порядку на сайте. Т.е. не KTB-02, если исходить из названия ветки для предыдущего ПР, а KTB-04
> 
> 1. Добавлен data class Word, хранящий слово, его перевод и количество правильных ответов 2) Добавлен список dictionary, хранящий объекты типа Word 3) Получение значения для количества правильных ответов реализовано в отдельном методе getCorrectAnswersCountValue. Учитывается не только что значения может не быть, но и то, что может быть пустая строка 4) Заполнение dictionary.
> 2. Вывод в консоль элементов из dictionary

а почему ты не присылаешь задания по порядку? до этого у тебя был только 1 мр с первым заданием, а сейчас сразу 4. получается остальные не будут зачтены. давай отправлять ПРы по порядку

DenisGubernatorov commented:
> > В ветке указан номер задачи, согласно порядку на сайте. Т.е. не KTB-02, если исходить из названия ветки для предыдущего ПР, а KTB-04
> > 
> > 1. Добавлен data class Word, хранящий слово, его перевод и количество правильных ответов 2) Добавлен список dictionary, хранящий объекты типа Word 3) Получение значения для количества правильных ответов реализовано в отдельном методе getCorrectAnswersCountValue. Учитывается не только что значения может не быть, но и то, что может быть пустая строка 4) Заполнение dictionary.
> > 2. Вывод в консоль элементов из dictionary
> 
> а почему ты не присылаешь задания по порядку? до этого у тебя был только 1 мр с первым заданием, а сейчас сразу 4. получается остальные не будут зачтены. давай отправлять ПРы по порядку
Получается, что KTB-01, KTB-03 это учебный материал и по ним не отправляется ПР. Я как раз и исходил из той логики, что в ПР название ветки соответсвует задаче в котором отправляем решение. В данном случае KTB-04. Если переименовывать названия  ветки здесь, то этот ПР закроется. Может тогда этот ПР закроем. Я удалю ветки с таким названием и создам новый ПР?
![image](https://github.com/user-attachments/assets/bbca4fb8-af6d-4688-ac3a-6b51f5c07727)

DenisGubernatorov committed:
KTB-04-string-parse
1) Рефакторинг кода. Убрана лишняя функция для получения количества правильных ответов
2) Переименование ветки согласно порядковому номеру

kaznikit commented:
> > > В ветке указан номер задачи, согласно порядку на сайте. Т.е. не KTB-02, если исходить из названия ветки для предыдущего ПР, а KTB-04
> > > 
> > > 1. Добавлен data class Word, хранящий слово, его перевод и количество правильных ответов 2) Добавлен список dictionary, хранящий объекты типа Word 3) Получение значения для количества правильных ответов реализовано в отдельном методе getCorrectAnswersCountValue. Учитывается не только что значения может не быть, но и то, что может быть пустая строка 4) Заполнение dictionary.
> > > 2. Вывод в консоль элементов из dictionary
> > 
> > 
> > а почему ты не присылаешь задания по порядку? до этого у тебя был только 1 мр с первым заданием, а сейчас сразу 4. получается остальные не будут зачтены. давай отправлять ПРы по порядку
> > Получается, что KTB-01, KTB-03 это учебный материал и по ним не отправляется ПР. Я как раз и исходил из той логики, что в ПР название ветки соответсвует задаче в котором отправляем решение. В данном случае KTB-04. Если переименовывать названия  ветки здесь, то этот ПР закроется. Может тогда этот ПР закроем. Я удалю ветки с таким названием и создам новый ПР?
> > ![image](https://private-user-images.githubusercontent.com/72883929/369407582-bbca4fb8-af6d-4688-ac3a-6b51f5c07727.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5Mjg5NTYsIm5iZiI6MTcyNjkyODY1NiwicGF0aCI6Ii83Mjg4MzkyOS8zNjk0MDc1ODItYmJjYTRmYjgtYWY2ZC00Njg4LWFjM2EtNmI1MWY1YzA3NzI3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDE0MjQxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThmMTk0ZDNhNmUwOWIxOTQ0NzAyMmQ0NWQ3ZmE1NDBiNWM4NGY5ZDQwOWVkMzg3NTAyNDllZmY0MzE3ODRiNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Q9ncRnRjaR0uL9pjoh8c_kZiraCMXgeTCYIDtvoomyw)

ок, сейчас аппрув, но следующее задание присылай именно для KTB-05. не нужно перескакивать, иначе проверять будет сложно из-за большого объема ПРа
